src/util/check-system.js

```diff
@@ -1,11 +1,44 @@
 import { exec } from 'child_process';
 import semver from 'semver';
 
-export default async () =>
-  new Promise((resolve) => {
+import { hasYarn, yarnOrNpmSpawn } from './yarn-or-npm';
+
+async function checkGitExists() {
+  return new Promise((resolve) => {
     exec('git --version', (err) => {
       if (err) return resolve(false);
       resolve(true);
     });
-  })
-  .then(prev => Promise.resolve(prev && semver.gt(process.versions.node, '6.0.0')));
+  });
+}
+
+async function checkNodeVersion() {
+  return Promise.resolve(semver.gt(process.versions.node, '6.0.0'));
+}
+
+const NPM_WHITELISTED_VERSIONS = '^3.0.0 || ^4.0.0 || ~5.1.0 || ~5.2.0 || >= 5.4.0';
+const YARN_WHITELISTED_VERSIONS = '0.23.3 || 0.24.6';
+
+function versionWarning(packageManager, whitelistedVersions) {
+  console.warning((`You are using ${packageManager}, but not a known good version. The known ` +
```

Comment:
In other places with warnings like this we have `${logSymbols.warning}` at the start of the log